src/Symfony/Component/Notifier/Bridge/Lox24/Lox24Transport.php

```diff
@@ -0,0 +1,204 @@
+        }
+
+        $from = $message->getFrom() ?: $this->from;
+
+        if (!$this->isFromValid($from)) {
+            throw new InvalidArgumentException(
+                "The \"From\" number \"$from\" is not a valid phone number, shortcode, or alphanumeric sender ID."
+            );
+        }
+
+
+        $body = [
+            'sender_id' => $from,
+            'phone' => $message->getPhone(),
+            'text' => $message->getSubject(),
+        ];
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $body = $this->setIsTextDeleted($body, $options);
+        $body = $this->setCallbackData($body, $options);
+        $body = $this->setDeliveryAt($body, $options);
+        $body = $this->setServiceCode($body, $options);
+        $body = $this->setVoiceLang($body, $options);
+
+        $response = $this->client->request('POST', "https://{$this->getEndpoint()}/sms", [
+            'headers' => [
+                'X-LOX24-AUTH-TOKEN' => $this->auth,
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+                'User-Agent' => 'LOX24 Symfony Notifier',
+            ],
+            'body' => $body,
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote LOX24 server.', $response, 0, $e);
+        }
+
+        if (201 !== $statusCode) {
+            $error = $response->toArray(false);
+
+            throw new TransportException(
+                "Unable to send the SMS: {$error['detail']}", $response
+            );
+        }
+
+        $success = $response->toArray(false);
+
+        $sentMessage = new SentMessage($message, (string)$this);
+        $sentMessage->setMessageId($success['uuid']);
+
+        return $sentMessage;
+    }
+
+
+    private function setVoiceLang(array $body, array $options): array
+    {
+        $voiceLang = $options['voice_lang'] ?? null;
+        if ($voiceLang) {
+            $voiceLang = strtolower($voiceLang);
+            if (!$this->isVoiceLangValid($voiceLang)) {
+                throw new InvalidArgumentException(
+                    "The \"voice_lang\" option \"$voiceLang\" is not a valid language. Allowed languages are: "
+                    .implode(', ', self::ALLOWED_VOICE_LANGUAGES)."."
+                );
+            }
+
+            if($voiceLang !== 'auto') {
+                $body['voice_lang'] = $voiceLang;
+            }
+        }
+
+        return $body;
+    }
+
+    private function isFromValid(string $from): bool
+    {
+        return preg_match('/^[.\-a-zA-Z0-9_ ]{2,11}$/', $from) || preg_match('/^\+[1-9]\d{1,14}$/', $from);
+    }
+
+    private function isVoiceLangValid(string $voiceLang): bool
+    {
+        return in_array($voiceLang, self::ALLOWED_VOICE_LANGUAGES, true);
+    }
+
+    private function setServiceCode(array $body, array $options): array
```

Comment:
What about adding an enum for ServiceCode and then add `getServiceCode` method to `Type` enum ?

Reply:
done